boot/bootutil/src/loader.c

```diff
@@ -541,6 +541,24 @@ boot_validate_slot(struct boot_loader_state *state, int slot,
         goto out;
     }
 
+#if defined(MCUBOOT_OVERWRITE_ONLY) && defined(MCUBOOT_DOWNGRADE_PREVENTION)
+    if (slot != BOOT_PRIMARY_SLOT) {
+        /* Check if version of secondary slot is sufficient */
+        rc = boot_is_version_sufficient(
+                &boot_img_hdr(state, BOOT_PRIMARY_SLOT)->ih_ver,
+                &boot_img_hdr(state, BOOT_SECONDARY_SLOT)->ih_ver);
+        if (rc != 0) {
+            BOOT_LOG_ERR("insufficient version in secondary slot");
+            flash_area_erase(fap, 0, fap->fa_size);
+            /* Image in the secondary slot does not satisfy version requirement.
+             * Erase the image and continue booting from the primary slot.
+             */
+            rc = -1;
```

Comment:
Shouldn't this be returning '1' to indicate that there is no image found? I'm not sure it affects the behavior much, but at least that is how I interpret the description in the header.

Reply:
Well, there is an image found, the same way in the case below, its just that its invalid. It is "bootable" however, its just that it doesn't satisfy all the criteria. IMO it should be -1 to be consistent with the signature check fail, but I can change it if you want to.

Reply:
I'm not sure it affects any behavior. But, there really isn't an image at that point, since it just got erased.

Reply:
Makes sense, changing to 1, should I change the other case as well, so that the behavior is consistent? The image in secondary is also erased in that case.

Reply:
Set both to 1. -1 is still used for the error case.